### DISCUSSION TITLE
pre-propulated db for web?

### DISCUSSION THREAD

**Discussion by kpk-i710** [simolus3/drift#2120] [Q&A]
I found this " You can use drift with a pre-propulated database that you ship with your app. This page also describes how to export the underlying sqlite3 database used by drift into a file. " does this one work for the web ? I could not

**Comment by simolus3** [comment]
You could manually construct the underlying storage mechanism, check if there's a database available locally and use the asset if there isn't. So something like this:

```dart
LazyDatabase(() async {
  final storage = DriftWebStorage(nameForDatabase);
  await storage.open();
  if (await storage.restore() == null) {
    // Download asset, use storage.store() to save it
  }
  await storage.close();
  return WebDatabase.withStorage(storage);
})
```

**Reply by kpk-i710** [reply]
thanks for the answer, I couldn't find the answer for a long time . I'll try as soon as possible
